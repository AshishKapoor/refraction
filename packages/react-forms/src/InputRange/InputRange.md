The `InputRange` component aims to provide an interface compatible with [`input[type=range]`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/range).

```js
<InputRange step={5} name="single1" onChange={console.log} />
```

All the properties supported by the HTML element are supported by `InputRange`, but,
additionally, this component supports two values at the same time.

The `onChange` handler in this case will return an object containing the properties
`start` and `end`, rather than a single number.

Note that in order to switch the component behavior to the double valued alternative
you must define a `defaultValue` defining the `start` and `end` properties, or define
its `value`.

```js
<InputRange
  defaultValue={{ start: 10, end: 40 }}
  name="double1"
  onChange={console.log}
/>
```

As with the HTML version, you can provide a `discrete` boolean property to display
some little ticks across the bar to show the valid and selectionable values.

```js
<InputRange step={5} defaultValue={0} name="single2" discrete />
```
