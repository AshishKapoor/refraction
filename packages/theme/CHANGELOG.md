# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.5.1](https://github.com/quid/ui-framework/compare/v1.5.0...v1.5.1) (2018-12-06)


### Bug Fixes

* use resolutions to point to microbundle fork ([16819ab](https://github.com/quid/ui-framework/commit/16819ab))





# [1.5.0](https://github.com/quid/ui-framework/compare/v1.4.0...v1.5.0) (2018-12-06)


### Bug Fixes

* use testPathPattern rather than runTestsByPath ([2350a80](https://github.com/quid/ui-framework/commit/2350a80))





# [1.4.0](https://github.com/quid/ui-framework/compare/v1.3.1...v1.4.0) (2018-12-05)

**Note:** Version bump only for package @quid/theme





# [1.3.0](https://github.com/quid/ui-framework/compare/v1.2.1...v1.3.0) (2018-12-05)


### Bug Fixes

* properly render CSS ([05f2920](https://github.com/quid/ui-framework/commit/05f2920))
* textStyles is not part of the theme ([919b026](https://github.com/quid/ui-framework/commit/919b026))


### Features

* new theme and react-forms packages ([61a44a0](https://github.com/quid/ui-framework/commit/61a44a0))
