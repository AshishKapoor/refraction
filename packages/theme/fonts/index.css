@import './ibmplex/index.css';
@import './icons/index.css';
