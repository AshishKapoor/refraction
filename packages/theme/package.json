{
  "name": "@quid/theme",
  "version": "1.5.2",
  "description": "Colors, fonts and other styling related resources, consumable by Emotion",
  "main": "dist/index.js",
  "main:umd": "dist/index.umd.js",
  "module": "dist/index.es.js",
  "license": "MIT",
  "scripts": {
    "start": "microbundle watch",
    "prepare": "microbundle build && flow-copy-source --ignore '{__mocks__/*,*.test}.js' src dist",
    "test": "cd ../.. && yarn test --testPathPattern packages/theme"
  },
  "dependencies": {
    "@emotion/core": "^10.0.2",
    "@emotion/utils": "^0.11.1",
    "color": "^3.1.0",
    "emotion-theming": "^10.0.2",
    "react": "^16.3.0"
  },
  "devDependencies": {
    "flow-copy-source": "^2.0.2",
    "microbundle": "^0.8.3"
  }
}
