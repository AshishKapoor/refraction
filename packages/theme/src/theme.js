/**
 * Copyright (c) Quid, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
// @flow
import colors from './colors';
import * as themes from './themes';
import sizes from './sizes';

const theme = {
  light: {
    current: 'light',
    colors,
    sizes,
    ...themes.light,
  },
  dark: {
    current: 'dark',
    colors,
    sizes,
    ...themes.dark,
  },
};

export default theme;
