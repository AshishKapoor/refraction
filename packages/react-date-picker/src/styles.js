// @flow
import styled from '@emotion/styled/macro';
import { textStyles, colors, withFallback as wf } from '@quid/theme';

export const CELL_HEIGHT = 30;
export const CELL_WIDTH = 30;

export const Container = styled.div`
  width: 330px;
  height: 422px;
  border-radius: 2px;
  display: flex;
  flex-direction: column;

  &:focus:not(:focus-visible) {
    outline: 0;
  }
`;

export const Navigator = styled.div`
  display: flex;
  align-items: center;
  padding-top: 10px;
  padding-bottom: 10px;
  color: ${wf(props => props.theme.primary)};
`;

export const Year = styled.span`
  ${textStyles('large')}
  margin-left: auto;
  margin-right: auto;
`;

export const Th = styled.th`
  font-weight: normal;
`;

export const Calendar = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
  color: ${colors.black};
  background-color: ${wf(props =>
    props.theme.current === 'light'
      ? props.theme.colors.white
      : props.theme.colors.gray6
  )};
  border: 1px solid ${wf(props => props.theme.colors.gray3)};
  border-top: 0px;
  border-radius: 0 0 2px 2px;
  padding: 0 6px 20px 6px;
`;

export const Table = styled.table`
  text-align: center;
`;

export const Days = styled.tbody`
  &::before {
    content: '';
    display: block;
    height: 3px;
  }
`;
