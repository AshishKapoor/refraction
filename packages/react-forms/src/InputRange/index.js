/**
 * Copyright (c) Quid, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
// @flow
import * as React from 'react';
import ResizeAware from 'react-resize-aware';
import styled from '@emotion/styled/macro';
import { withFallback as wf } from '@quid/theme';
import {
  TRACK_BACKGROUND,
  Container,
  Track,
  Segment,
  StepTicks,
} from './styles';
import Handle from './Handle';

export type ComputedValue = { start: number, end: number };
export type Value = number | { start?: number, end?: number };

// We need this to give a proper default value to `track`
// otherwise during tests something may fail because of the
// jsdom quirks that make the component lifecycle execute without
// having the root node initialized
const div = document.createElement('div');

export const handles = {
  START: 'start',
  END: 'end',
};

export const keys = {
  RIGHT: 39,
  LEFT: 37,
};

export function fromPercentToPixel(percent: number, width: number): number {
  return (width / 100) * percent;
}

function hasName(name): boolean %checks {
  return typeof name === 'string';
}

// We used to use CSS gradients with hard stops, but this bug is
// producing a bad visual result (crbug.com/696603)
export function wrapWithSvg(
  content: React$Element<any>,
  { className, on, ...props }: { className?: string, on: boolean }
) {
  return (
    <svg
      className={className}
      xmlns="http://www.w3.org/2000/svg"
      width="100"
      height="100"
      {...props}
    >
      {content}
    </svg>
  );
}

export const Rect = styled(props => (
  <rect
    {...props}
    fill="currentColor"
    height={2}
    width={Math.max(props.width, 0)}
    x={Math.max(props.offset, 0)}
    rx={1}
    ry={1}
  />
))`
  color: ${wf(props =>
    props.disabled ? TRACK_BACKGROUND(props) : props.theme.selected
  )};
`;

export function drawSegment(width: number, offset: number, disabled: boolean) {
  return (props: Object) =>
    wrapWithSvg(
      <Rect width={width} offset={offset} disabled={disabled} />,
      props
    );
}

export function getDecimals(num: number) {
  const stepArr = num.toString().split('.');
  return stepArr[1] ? stepArr[1].length : 0;
}

export function toFixed(num: number, decimals: number) {
  const parts = num.toString().split('.');
  return `${parts[0]}.${(parts[1] || '00000000').slice(0, decimals)}`;
}

export function valueToObject(value?: Value): Object {
  if (value === undefined) {
    return {};
  }

  if (typeof value === 'object') {
    if (
      value.hasOwnProperty(handles.START) &&
      !isNaN(value[handles.START]) &&
      value.hasOwnProperty(handles.END) &&
      !isNaN(value[handles.END])
    ) {
      return {
        [handles.START]: +value[handles.START],
        [handles.END]: +value[handles.END],
      };
    } else if (
      value.hasOwnProperty(handles.START) &&
      !isNaN(value[handles.START])
    ) {
      return {
        [handles.START]: +value[handles.START],
      };
    } else {
      return {
        [handles.END]: +value[handles.END],
      };
    }
  }

  return {
    [handles.START]: +value,
  };
}

export function computeW3cDefaultValue(min: number, max: number) {
  return max > min ? min + (max - min) / 2 : min;
}

export function isItTwoHandled(value: ComputedValue) {
  return !isNaN(value[handles.END]);
}

export function areValuesEqual(a?: Value, b?: Value): boolean {
  const compA = valueToObject(a);
  const compB = valueToObject(b);

  return (
    compA[handles.START] === compB[handles.START] &&
    compA[handles.END] === compB[handles.END]
  );
}

type Props = {
  onChange?: Function,
  value?: Value,
  defaultValue?: Value,
  name?: string,
  min: number,
  max: number,
  step: number,
  readOnly: boolean,
  disabled: boolean,
  discrete: boolean,
  className?: string,
};

type Updater = () => Object;
type HandleType = 'start' | 'end';
type State = {
  width: number,
  value?: Object,
  dragging: boolean,
  activeHandle: HandleType,
  stepLen: number,
  fontSize: number,
};

// The following component relies on a lot of DOM APIs to work, I'm not able
// to test it with Jest and JSDOM unfortunately
/* istanbul ignore next */
class Inner extends React.PureComponent<Props, State> {
  track: HTMLElement = div;

  static defaultProps = {
    min: 0,
    max: 100,
    step: 1,
    readOnly: false,
    disabled: false,
    discrete: false,
  };

  // Don't use state because
  // 1. We don't care/want to re-render on each pos update
  // 2. We need this value to be updated istantly
  pos = 0;

  state = {
    width: 0,
    value: undefined,
    dragging: false,
    activeHandle: handles.START,
    stepLen: 0,
    fontSize: 13,
  };

  uncontrolledOnChange = (updater: Updater) => {
    const prevValue = { ...this.state.value };
    this.setState(updater, () => {
      const isTwoHandled = this.state.value
        ? isItTwoHandled(this.state.value)
        : false;
      if (this.props.onChange && !areValuesEqual(prevValue, this.state.value)) {
        this.props.onChange(
          // $FlowFixMe(fzivolo): need better type checking for `isTwoHandled`, just being lazy
          isTwoHandled ? this.state.value : this.state.value.start
        );
      }
    });
  };

  controlledOnChange = (updater: Updater) => {
    const { value } = updater();
    const isTwoHandled = isItTwoHandled(value);
    const newValue = isTwoHandled ? value : value[handles.START];
    if (this.props.onChange && !areValuesEqual(this.props.value, newValue)) {
      this.props.onChange(isTwoHandled ? newValue : newValue.start);
    }
  };

  get onChange(): Updater => void {
    return this.props.onChange &&
      (!isNaN(valueToObject(this.props.value)[handles.START]) ||
        !isNaN(valueToObject(this.props.value)[handles.END]))
      ? this.controlledOnChange
      : this.uncontrolledOnChange;
  }

  handleMouseDown = (
    { clientX: pos }: { clientX: number },
    activeHandle: HandleType
  ) => {
    if (this.props.disabled || this.props.readOnly) {
      return;
    }

    this.updatePos(pos);
    this.setState({ dragging: true, activeHandle });
  };

  handleMouseMove = (evt: MouseEvent | Object) => {
    const { clientX: curPos } = evt;
    const {
      track,
      pos,
      state: { dragging, stepLen, activeHandle },
    } = this;

    if (track && dragging) {
      const { left } = track.getBoundingClientRect();

      // This is used to avoid unwanted text selection while dragging
      evt.preventDefault && evt.preventDefault();

      const diff = curPos - pos;
      const change = diff / stepLen;

      const currentValue = this.computedValue[activeHandle];
      const newValue =
        Math.round((currentValue + change) / this.props.step) * this.props.step;

      this.updatePos(left + (newValue - this.props.min) * stepLen);
      this.updateValue(newValue);
    }
  };

  handleMouseUp = () => this.setState({ dragging: false });

  handleTrackMouseDown = ({ clientX }: { clientX: number }) => {
    if (this.props.disabled || this.props.readOnly) {
      return;
    }

    const { min } = this.props;
    const { stepLen } = this.state;
    const { left } = this.track.getBoundingClientRect();
    const value = this.computedValue;
    const startOffset = stepLen * value[handles.START];
    const endOffset = stepLen * value[handles.END];
    const middleOffset = startOffset + (endOffset - startOffset) / 2;
    const curPos = clientX - left;

    let activeHandle;
    if (curPos > middleOffset && value[handles.END]) {
      activeHandle = handles.END;
    } else {
      activeHandle = handles.START;
    }
    this.updatePos(left + (value[activeHandle] - min) * stepLen);

    this.setState({ dragging: true, activeHandle }, () => {
      this.handleMouseMove({ clientX });
    });
  };

  handleKeyDown = ({ which }: KeyboardEvent) => {
    if (this.props.disabled || this.props.readOnly) {
      return;
    }

    const { activeHandle } = this.state;
    const { step } = this.props;

    if (which === keys.LEFT) {
      this.updateValue(this.computedValue[activeHandle] - step);
    } else if (which === keys.RIGHT) {
      this.updateValue(this.computedValue[activeHandle] + step);
    }
  };

  updateValue = (newValue: number) =>
    this.onChange((state = this.state) => {
      const safeNewValue =
        Math.round(
          parseFloat(toFixed(newValue, getDecimals(this.props.step))) /
            this.props.step
        ) * this.props.step;

      const { activeHandle, value = this.computedValue } = state;
      const { min, max } = this.props;

      const activeHandleValue =
        activeHandle === handles.START
          ? Math.min(
              !isNaN(value[handles.END]) ? value[handles.END] : max,
              Math.max(min, safeNewValue)
            )
          : Math.max(Math.min(max, safeNewValue), value[handles.START]);

      return {
        value: {
          ...value,
          [activeHandle]: activeHandleValue,
        },
      };
    });

  updatePos = (pos: number) => {
    const { left, right } = this.track.getBoundingClientRect();
    this.pos = Math.max(Math.min(pos, right), left);
  };

  computeSizes = () => {
    this.setState((state, { min, max }) => ({
      width: this.track.offsetWidth,
      stepLen: this.track.offsetWidth / (max - min),
      fontSize: Number(
        window
          .getComputedStyle(this.track)
          .getPropertyValue('font-size')
          .replace('px', '')
      ),
    }));
  };

  get computedValue(): ComputedValue {
    const { value, defaultValue, min, max } = this.props;

    const stateValueObj = valueToObject(this.state.value);
    const propsValueObj = valueToObject(value);
    const defaultValueObj = valueToObject(defaultValue);
    const w3cDefaultValue = valueToObject(computeW3cDefaultValue(min, max));

    // If it's a controlled input, use `value`
    // If there's already a `value` in state, use it
    // If it's an uncontrolled input, use `defaultValue` as initial value
    // If nothing is defined, we fallback to the default defined by W3C
    return {
      [handles.START]: !isNaN(propsValueObj[handles.START])
        ? propsValueObj[handles.START]
        : !isNaN(stateValueObj[handles.START])
        ? stateValueObj[handles.START]
        : !isNaN(defaultValueObj[handles.START])
        ? defaultValueObj[handles.START]
        : w3cDefaultValue[handles.START],
      [handles.END]: !isNaN(propsValueObj[handles.END])
        ? propsValueObj[handles.END]
        : !isNaN(stateValueObj[handles.END])
        ? stateValueObj[handles.END]
        : !isNaN(defaultValueObj[handles.END])
        ? defaultValueObj[handles.END]
        : w3cDefaultValue[handles.END],
    };
  }

  getSegments = (isTwoHandled: boolean) => {
    const { min, max, disabled, readOnly } = this.props;
    const { width, fontSize } = this.state;
    const isGrayedOut = disabled || readOnly;
    const computedValue = this.computedValue;
    const { start, end = max } = computedValue;
    const unit = 100 / (max - min);

    // 0.43em is the handle size as defined in the CSS of this component
    const handleSize = fontSize * 0.462;
    const padding = isGrayedOut ? handleSize : 0;

    const a: number = isTwoHandled ? start - min : 0;
    const b: number = isTwoHandled ? end - min : start - min;

    const seg1W = fromPercentToPixel(unit * a, width);
    const Seg1 = drawSegment(seg1W - padding, 0, true);
    const seg2W =
      fromPercentToPixel(unit * b, width) -
      fromPercentToPixel(unit * a, width) -
      padding * (isTwoHandled ? 2 : 1);
    const seg2O = seg1W + (isTwoHandled ? padding : 0);
    const Seg2 = drawSegment(seg2W, seg2O, false);
    const Seg3 = drawSegment(
      fromPercentToPixel(100 - unit * b, width) - padding,
      fromPercentToPixel(unit * b, width) + padding,
      true
    );

    return [
      isTwoHandled && <Segment as={Seg1} key="1" />,
      <Segment as={Seg2} key="2" on />,
      <Segment as={Seg3} key="3" />,
    ];
  };

  // start and end refer to the handle's name
  handleFocusStart = () => this.setState({ activeHandle: handles.START });
  handleFocusEnd = () => this.setState({ activeHandle: handles.END });
  handleMouseDownStart = (evt: MouseEvent) =>
    this.handleMouseDown(evt, handles.START);
  handleMouseDownEnd = (evt: MouseEvent) =>
    this.handleMouseDown(evt, handles.END);

  componentDidMount() {
    this.computeSizes();
    document.addEventListener('mousemove', this.handleMouseMove);
    document.addEventListener('mouseup', this.handleMouseUp);
  }

  componentWillUnmount() {
    document.removeEventListener('mousemove', this.handleMouseMove);
    document.removeEventListener('mouseup', this.handleMouseUp);
  }

  render() {
    const {
      computedValue,
      props: {
        discrete,
        name,
        min,
        max,
        readOnly,
        disabled,
        onChange, // eslint-disable-line no-unused-vars
        step,
        className,
        ...props
      },
      state: { dragging, activeHandle },
    } = this;

    const isGrayedOut = readOnly || disabled;

    // Is two handled only if it has both `start` and `end` props
    const isTwoHandled = isItTwoHandled(computedValue);
    const valueText = isTwoHandled
      ? `${computedValue[handles.START]} - ${computedValue[handles.END]}`
      : `${computedValue[handles.START]}`;

    // If there's a single handle, we keep the original name
    // in case there are two handles, we append `1` and `2` to the original name
    // to distinguish them
    let nameStart, nameEnd;
    if (hasName(name)) {
      nameStart = isTwoHandled ? `${name}1` : name;
      nameEnd = `${name}2`;
    }

    // Ticks shouldn't overlap with handles, we hide them if necessary
    const hiddenTicks = Object.values(computedValue)
      .map(
        (value): ?number =>
          typeof value === 'number' ? (value - min) / step : undefined
      )
      .filter(value => value !== undefined);

    return (
      <Container
        as={ResizeAware}
        role="slider"
        aria-valuemin={min}
        aria-valuemax={max}
        aria-valuetext={valueText}
        aria-readonly={readOnly}
        aria-disabled={disabled}
        aria-orientation="horizontal"
        onResize={this.computeSizes}
        {...props}
      >
        <Track
          disabled={isGrayedOut}
          ref={el => el !== null && (this.track = el)}
          onMouseDown={this.handleTrackMouseDown}
          data-action="click-track"
        >
          {this.getSegments(isTwoHandled)}
        </Track>
        <Handle
          value={computedValue[handles.START]}
          name={hasName(name) ? nameStart : undefined}
          disabled={disabled}
          readOnly={readOnly}
          dragging={activeHandle === handles.START && dragging}
          onMouseDown={this.handleMouseDownStart}
          onKeyDown={this.handleKeyDown}
          offset={Math.round(
            ((computedValue[handles.START] - min) / (max - min)) *
              this.state.width
          )}
          onFocus={this.handleFocusStart}
        />
        {discrete && (
          <StepTicks step={step} min={min} max={max} hidden={hiddenTicks} />
        )}
        {isTwoHandled && (
          <Handle
            value={computedValue[handles.END]}
            name={hasName(name) ? nameEnd : undefined}
            disabled={disabled}
            readOnly={readOnly}
            dragging={activeHandle === handles.END && dragging}
            onMouseDown={this.handleMouseDownEnd}
            onKeyDown={this.handleKeyDown}
            offset={Math.round(
              ((computedValue[handles.END] - min) / (max - min)) *
                this.state.width
            )}
            onFocus={this.handleFocusEnd}
          />
        )}
      </Container>
    );
  }
}

const InputRange = styled(Inner)();

// @component
export default InputRange;
