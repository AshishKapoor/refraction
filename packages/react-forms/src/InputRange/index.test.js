// @flow
import * as React from 'react';
import { mount } from 'enzyme';
import { colors } from '@quid/theme';
import InputRange, {
  getDecimals,
  toFixed,
  valueToObject,
  computeW3cDefaultValue,
  areValuesEqual,
  Rect,
} from '.';
import { Track, TRACK_BACKGROUND } from './styles';
import Handle from './Handle';

it('renders an InputRange', () => {
  expect(
    mount(
      <InputRange
        step={5}
        min={0}
        max={10}
        defaultValue={0}
        name="x"
        discrete
      />
    )
  ).toMatchSnapshot();
});

it('renders a Track component', () => {
  expect(mount(<Track />)).toMatchSnapshot();
  expect(mount(<Track disabled />)).toMatchSnapshot();
});

it('renders a disabled Handle', () => {
  expect(
    mount(<Handle disabled />)
      .find('input')
      .prop('disabled')
  ).toBeTruthy();
});

it('renders a zero Handle', () => {
  expect(mount(<Handle zero />)).toHaveStyleRule(
    'border',
    `1px solid ${colors.gray4}`
  );
});

it('renders a dragging Handle', () => {
  expect(mount(<Handle dragging />)).toHaveStyleRule('transition-delay', `0s`);
});

it('returns the alt background on dark theme', () => {
  expect(
    TRACK_BACKGROUND({ theme: { current: 'dark', colors: { gray2: 'gray2' } } })
  ).toMatchInlineSnapshot(`"gray2"`);
});

it('tests getDecimals', () => {
  expect(getDecimals(10.123)).toBe(3);
  expect(getDecimals(10)).toBe(0);
});

it('tests toFixed', () => {
  expect(toFixed(10.1234, 2)).toBe('10.12');
  expect(toFixed(10, 2)).toBe('10.00');
});

it('tests valueToObject', () => {
  expect(valueToObject(10)).toMatchInlineSnapshot(`
Object {
  "start": 10,
}
`);
  expect(valueToObject({ start: 5 })).toMatchInlineSnapshot(`
Object {
  "start": 5,
}
`);
  expect(valueToObject({ start: 5, end: 10 })).toMatchInlineSnapshot(`
Object {
  "end": 10,
  "start": 5,
}
`);
  expect(valueToObject({ end: 10 })).toMatchInlineSnapshot(`
Object {
  "end": 10,
}
`);
});

it('tests computeW3cDefaultValue', () => {
  expect(computeW3cDefaultValue(0, 10)).toBe(5);
  expect(computeW3cDefaultValue(10, 0)).toBe(10);
});

it('tests areValuesEqual', () => {
  expect(areValuesEqual(1, 1)).toBeTruthy();
  expect(areValuesEqual(1, 2)).toBeFalsy();
});

it('renders Rect', () => {
  // $FlowIgnoreMe
  console.error = jest.fn();
  expect(mount(<Rect width={-1} />)).toHaveStyleRule('width', '0px');
});
