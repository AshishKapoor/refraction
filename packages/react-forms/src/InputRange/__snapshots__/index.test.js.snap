// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Track component 1`] = `
.emotion-0 {
  position: relative;
  top: 50%;
  background-position: center;
  background-size: 100% 2px;
  background-repeat: no-repeat;
  height: 2px;
  border-radius: 2px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.emotion-0::before {
  content: '';
  position: absolute;
  height: 6px;
  width: 100%;
  -webkit-transform: translateY(-25%);
  -ms-transform: translateY(-25%);
  transform: translateY(-25%);
}

<Track>
  <div
    className="emotion-0 emotion-1"
  />
</Track>
`;

exports[`renders a Track component 2`] = `
.emotion-0 {
  position: relative;
  top: 50%;
  background-position: center;
  background-size: 100% 2px;
  background-repeat: no-repeat;
  height: 2px;
  border-radius: 2px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.emotion-0::before {
  content: '';
  position: absolute;
  height: 4px;
  width: 100%;
  -webkit-transform: translateY(-25%);
  -ms-transform: translateY(-25%);
  transform: translateY(-25%);
}

<Track
  disabled={true}
>
  <div
    className="emotion-0 emotion-1"
    disabled={true}
  />
</Track>
`;

exports[`renders an InputRange 1`] = `
@keyframes animation-0 {
  from {
    box-shadow: 0 0 0 0 rgba(0,167,184,0.26);
  }

  to {
    box-shadow: 0 0 0 3px rgba(0,167,184,0.26);
  }
}

.emotion-22 {
  position: relative;
  height: 12px;
  isolation: isolate;
}

.emotion-8 {
  position: relative;
  top: 50%;
  background-position: center;
  background-size: 100% 2px;
  background-repeat: no-repeat;
  height: 2px;
  border-radius: 2px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.emotion-8::before {
  content: '';
  position: absolute;
  height: 6px;
  width: 100%;
  -webkit-transform: translateY(-25%);
  -ms-transform: translateY(-25%);
  transform: translateY(-25%);
}

.emotion-2 {
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  right: 0;
  height: 100%;
  width: 100%;
  color: #6C7983;
}

.emotion-0 {
  fill: #00c1bb;
  height: 2px;
  width: 0px;
  x: 0px;
  rx: 1;
  ry: 1;
}

.emotion-6 {
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  right: 0;
  height: 100%;
  width: 100%;
  color: #00c1bb;
}

.emotion-4 {
  fill: #6C7983;
  height: 2px;
  width: 0px;
  x: 0px;
  rx: 1;
  ry: 1;
}

.emotion-10 {
  color: #00c1bb;
  top: 50%;
  left: 0;
  position: absolute;
  width: 6px;
  height: 6px;
  margin-top: -3px;
  cursor: default;
  z-index: 2;
  -webkit-transform: translateX(calc(0px - 50%));
  -ms-transform: translateX(calc(0px - 50%));
  transform: translateX(calc(0px - 50%));
}

.emotion-10::before {
  content: '';
  display: block;
  background-color: currentColor;
  width: 6px;
  height: 6px;
  -webkit-transform: scale(2);
  -ms-transform: scale(2);
  transform: scale(2);
  border-radius: 50%;
  -webkit-transition: -webkit-transform 0.2s ease-in-out,width 0.2s ease-in-out,height 0.2s ease-in-out;
  -webkit-transition: transform 0.2s ease-in-out,width 0.2s ease-in-out,height 0.2s ease-in-out;
  transition: transform 0.2s ease-in-out,width 0.2s ease-in-out,height 0.2s ease-in-out;
}

.emotion-10:active::before {
  -webkit-transform: scale(3);
  -ms-transform: scale(3);
  transform: scale(3);
  -webkit-transition-delay: 0s;
  transition-delay: 0s;
}

.emotion-10:focus {
  outline: 0;
}

.emotion-10:focus-visible::before {
  -webkit-animation: animation-0 1s infinite alternate ease-in-out;
  animation: animation-0 1s infinite alternate ease-in-out;
}

.emotion-20 {
  pointer-events: none;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-12 {
  height: 6px;
  width: 2px;
  position: relative;
  -webkit-transform: translateX(1px) translateY(3px);
  -ms-transform: translateX(1px) translateY(3px);
  transform: translateX(1px) translateY(3px);
  color: #00c1bb;
  border-radius: 2px;
  background-color: #6C7983;
  opacity: 0;
}

.emotion-12:first-of-type {
  -webkit-transform: translateY(3px);
  -ms-transform: translateY(3px);
  transform: translateY(3px);
}

.emotion-14 {
  height: 6px;
  width: 2px;
  position: relative;
  -webkit-transform: translateX(1px) translateY(3px);
  -ms-transform: translateX(1px) translateY(3px);
  transform: translateX(1px) translateY(3px);
  color: #00c1bb;
  border-radius: 2px;
  background-color: #6C7983;
  opacity: 1;
}

.emotion-14:first-of-type {
  -webkit-transform: translateY(3px);
  -ms-transform: translateY(3px);
  transform: translateY(3px);
}

<InputRange
  defaultValue={0}
  disabled={false}
  discrete={true}
  max={10}
  min={0}
  name="x"
  readOnly={false}
  step={5}
>
  <Container
    aria-disabled={false}
    aria-orientation="horizontal"
    aria-readonly={false}
    aria-valuemax={10}
    aria-valuemin={0}
    aria-valuetext="0"
    as={[Function]}
    defaultValue={0}
    onResize={[Function]}
    role="slider"
  >
    <div
      aria-disabled={false}
      aria-orientation="horizontal"
      aria-readonly={false}
      aria-valuemax={10}
      aria-valuemin={0}
      aria-valuetext="0"
      className="emotion-22 emotion-23"
      defaultValue={0}
      role="slider"
    >
      <object
        aria-hidden={true}
        onLoad={[Function]}
        style={
          Object {
            "display": "block",
            "height": "100%",
            "left": 0,
            "overflow": "hidden",
            "pointerEvents": "none",
            "position": "absolute",
            "top": 0,
            "visibility": "hidden",
            "width": "100%",
            "zIndex": -1,
          }
        }
        tabIndex={-1}
        type="text/html"
      />
      <Track
        data-action="click-track"
        disabled={false}
        key=".0"
        onMouseDown={[Function]}
      >
        <div
          className="emotion-8 emotion-9"
          data-action="click-track"
          disabled={false}
          onMouseDown={[Function]}
        >
          <Segment
            as={[Function]}
            key="2"
            on={true}
          >
            <svg
              className="emotion-2 emotion-3"
              height="100"
              width="100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <Rect
                disabled={false}
                offset={0}
                width={0}
              >
                <rect
                  className="emotion-0 emotion-1"
                  disabled={false}
                  offset={0}
                  width={0}
                />
              </Rect>
            </svg>
          </Segment>
          <Segment
            as={[Function]}
            key="3"
          >
            <svg
              className="emotion-6 emotion-3"
              height="100"
              width="100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <Rect
                disabled={true}
                offset={0}
                width={0}
              >
                <rect
                  className="emotion-4 emotion-1"
                  disabled={true}
                  offset={0}
                  width={0}
                />
              </Rect>
            </svg>
          </Segment>
        </div>
      </Track>
      <Handle
        disabled={false}
        dragging={false}
        key=".1"
        name="x"
        offset={0}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        readOnly={false}
        value={0}
      >
        <div
          className="emotion-10 emotion-11"
          data-action="drag-handle"
          data-value={0}
          offset={0}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          tabIndex={0}
        >
          <input
            disabled={false}
            name="x"
            readOnly={false}
            type="hidden"
            value={0}
          />
        </div>
      </Handle>
      <StepTicks
        hidden={
          Array [
            0,
          ]
        }
        key=".2"
        max={10}
        min={0}
        step={5}
      >
        <div
          className="emotion-20 emotion-21"
        >
          <Tick
            hidden={true}
            key="0"
          >
            <div
              className="emotion-12 emotion-13"
              hidden={true}
            />
          </Tick>
          <Tick
            hidden={false}
            key="1"
          >
            <div
              className="emotion-14 emotion-13"
              hidden={false}
            />
          </Tick>
          <Tick
            hidden={false}
            key="2"
          >
            <div
              className="emotion-14 emotion-13"
              hidden={false}
            />
          </Tick>
          <Tick
            hidden={false}
            key="3"
          >
            <div
              className="emotion-14 emotion-13"
              hidden={false}
            />
          </Tick>
        </div>
      </StepTicks>
    </div>
  </Container>
</InputRange>
`;
